src/main/java/com/yy5/employee/controller/EmployeeController.java

```diff
@@ -46,10 +44,10 @@ public ResponseEntity<Map<String, String>> handleEmployeeNotFoundException(
         return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
     }
     @PostMapping("/employees")
-    public ResponseEntity<EmployeeResponse> insert(@RequestBody EmployeeRequest employeeRequest, UriComponentsBuilder uriBuilder) {
-        Employee employee = employeeService.insert(employeeRequest.getName(),employeeRequest.getAge());
-        URI location = uriBuilder.path("employees/{employeeNumber}").buildAndExpand(employee.getEmployeeNumber()).toUri();
-        EmployeeResponse body = new EmployeeResponse("employee created");
-        return ResponseEntity.created(location).body(body);
+    public ResponseEntity<?> insert(@RequestBody @Validated EmployeeRequest employeeRequest, UriComponentsBuilder uriBuilder) throws MethodArgumentNotValidException {
```

Comment:
? を使う必要はないかと思います。
あと throws もなくてよいかと。
```suggestion
    public ResponseEntity<EmployeeResponse> insert(@RequestBody @Validated EmployeeRequest employeeRequest, UriComponentsBuilder uriBuilder)  {
```
なんで throws がなくていいのかなど気になったら考えてみてください。

Reply:
？はEmployeeResponseに修正しました。
>  なんで throws がなくていいのかなど気になったら考えてみてください。

`throws MethodArgumentNotValidException`を削除した場合、
`/Users/kohe/RaiseTech/employee/src/main/java/com/yy5/employee/controller/EmployeeController.java:50: エラー: 例外MethodArgumentNotValidExceptionは報告されません。スローするには、捕捉または宣言する必要があります
            Employee employee = employeeService.insert(employeeRequest.getName(),employeeRequest.getAge());`
というエラーメッセージが表示され、アプリケーションの立ち上げができません。

<img width="1373" alt="スクリーンショット 2024-02-04 5 17 58" src="https://github.com/YYprogramer/employee/assets/131517182/dcdded3d-8d33-487a-b74a-d3fa15f37a6c">

Reply:
Serviceクラスの修正を行なったら上記のエラーが修正されました。[62cdcb0](https://github.com/YYprogramer/employee/pull/12/commits/62cdcb083238bd1310bbea07f52e0eabb7b1a144)
[58c372d](https://github.com/YYprogramer/employee/pull/12/commits/58c372d44554a067427d6e4e1488979603c3cc50)

`throws MethodArgumentNotValidException`を明示しなくても良い理由は
@Validatedを定義した場合、デフォルトでSpringが例外処理をハンドリングしてくれるため、わざわざ`throws MethodArgumentNotValidException`を記述しなくても良い。ということで間違いないでしょうか？

Reply:
> throws MethodArgumentNotValidExceptionを明示しなくても良い理由は
@validatedを定義した場合、デフォルトでSpringが例外処理をハンドリングしてくれるため、わざわざthrows MethodArgumentNotValidExceptionを記述しなくても良い。ということで間違いないでしょうか？

おっしゃるとおりです。
厳密に言うとControllerに到達するまえにSpringがValidatedアノテーションのついたクラスの値をバリデーションしてMethodArgumentNotValidExceptionをthrowしてくれる。
したがって、実装者はMethodArgumentNotValidExceptionをthrowするというようなケアが不要になります。